src/System.IO.FileSystem/tests/FileStream/WriteByte.cs

```diff
@@ -59,5 +59,29 @@ public void WriteFile()
                 Assert.Equal(TestBuffer, buffer);
             }
         }
+
+        [Fact]
+        public void WriteFile_PastBufferSize()
+        {
+            string fileName = GetTestFilePath();
+            using (FileStream fs = new FileStream(fileName, FileMode.Create))
+            {
+                for (int written = 0; written < TestBuffer.Length * 1000; written++)
+                {
+                    fs.WriteByte(TestBuffer[written % TestBuffer.Length]);
+                }
+            }
+
+            using (FileStream fs = new FileStream(fileName, FileMode.Open))
+            {
+                byte[] buffer = new byte[TestBuffer.Length];
+                int offset = 0;
+                while (fs.Read(buffer, 0, TestBuffer.Length) > 0)
+                {
+                    offset += TestBuffer.Length;
+                    Assert.Equal(TestBuffer, buffer);
```

Comment:
It's technically possible this assert could fail, as Read is not required to return the full amount requested.  That said, both our Windows and Unix implementations try to.